src/forcemove.h

```diff
@@ -0,0 +1,98 @@
+#pragma once
+#include "move.h"
+#include "energy.h"
+#include "units.h"
+
+namespace Faunus {
+
+namespace Move {
+
+/**
+ * @brief Base class for dynamics integrators
+ *
+ */
+class IntegratorBase {
+  protected:
+    Space &spc;
+    Energy::Energybase &energy;
+    IntegratorBase(Space &spc, Energy::Energybase &energy);
+    virtual ~IntegratorBase() = default;
+
+  public:
+    /**
+     * @brief Move particles by one time step and update their positions, velocities and acting forces
+     * @param velocities
+     * @param forces
+     */
+    virtual void step(PointVector &velocities, PointVector &forces) = 0; // todo shall we return real time progress?
+    virtual void from_json(const json &j) = 0;
+    virtual void to_json(json &j) const = 0;
+};
+
+void from_json(const json &j, IntegratorBase &i);
+void to_json(json &j, const IntegratorBase &i);
+
+/**
+ * @brief Symmetric Langevin velocity-Verlet method (BAOAB)
+ *
+ * The integrator can conduct normal velocity-Verlet, when gamma = 0.
+ */
+class LangevinVelocityVerlet : public IntegratorBase {
+    float dt;    //!< integration time step (picoseconds)
+    float gamma; //!< friction coefficient (inverse picoseconds)
+
+    //! increment particle's position in an A semi-step
+    inline Point positionIncrement(const Point& velocity);
```

Comment:
put doxygen comment _after declaration, i.e. `; //!< increment particle's position in an A semi-step`. Remove variable name

Reply:
Here, the doxygen comment before the declaration helps keeping the rhythm when reading through the header file, as other comments in the same section are longer and before the methods' declarations.

Parameter names in method declarations. I am slowly building an opinion to rather keep them unless obvious and distracting (e.g., in `to_json` and `from_json` methods). It is true that the parameters' names in declarations have no purpose (are redundant) from the compiler's point of view. You also have to make an extra effort to keep them in accord with methods' definitions. However the advantage of more readable code outweighs this tiny need of extra care. Compare for example `void step(PointVector &, PointVector &)` and `void step(PointVector &velocities, PointVector &forces)`. In the former case, you have to jump to the definition to see how to use the method which is inconvenient at least. Are we following any particular style guide in this matter? Here are some additional arguments to support my (still weak) opinion on this.
https://stackoverflow.com/questions/7891526/should-function-declarations-include-parameter-names
https://stackoverflow.com/questions/771492/c-style-convention-parameter-names-within-class-declaration

Reply:
To support your case, I actually think CLion signature refactoring could get confused if names are missing in the declaration. I may be wrong, or perhaps it's a matter of settings. In any case, I don't have strong opinions about this, but have seen some cleaning tools complaining about names in headers.